### PR TITLE
Increased timeout value for delete history event task

### PR DESCRIPTION
**Pull request by timl3136**
**What changed?**
We increased the timeout value for delete history event task to dynamic config with default at 30s.

**Why?**
The timeout value for delete history event is too low causing large history deletion to constantly timeout

**How did you test it?**

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/task/timer_standby_task_executor.go [hunk 1]
```diff
@@ -110,7 +110,10 @@ func (t *timerStandbyTaskExecutor) Execute(
 	case persistence.TaskTypeWorkflowBackoffTimer:
 		return t.executeWorkflowBackoffTimerTask(ctx, timerTask)
 	case persistence.TaskTypeDeleteHistoryEvent:
-		return t.executeDeleteHistoryEventTask(ctx, timerTask)
+		// special timeout for delete history event
+		deleteHistoryEventContext, deleteHistoryEventCancel := context.WithTimeout(context.Background(), taskDeleteHistoryEventContextTimeout)
```
Comment (taylanisikdemir) [inline service/history/task/timer_standby_task_executor.go:114]: I know the existing `ctx` has 3 sec timeout and doesn't listen for queue shut down signals. 
With the new 60s timeout for history deletions it is going to be more likely to end up hanging a shutdown. We should consider making these operations responsive to shut down signals.
@Shaddoll what do you think?

Comment (taylanisikdemir) [reply service/history/task/timer_standby_task_executor.go:114]: Discussed offline. We will hook up Stop signals to task executor in a follow up PR. 
In this PR, let's make this 60s dynamically configurable

**Review by davidporter-id-au — APPROVED** [review]

File: common/dynamicconfig/constants.go [hunk 2]
```diff
@@ -1440,6 +1440,13 @@ const (
 	IsolationGroupStateUpdateRetryAttempts
 
 	LargeShardHistoryBlobMetricThreshold
+
+	// DeleteHistoryEventContextTimeout
```
Comment (taylanisikdemir) [inline common/dynamicconfig/constants.go:1444]: let's mention the unit of time in comment or variable name.

**Review by taylanisikdemir — APPROVED** [review]

File: common/dynamicconfig/constants.go (hunk 2 above)
Comment (timl3136) [reply common/dynamicconfig/constants.go:1444]: Thanks, I added that in L3803 but forgot to add it here.

**Pull request merged**